mrcnn/model.py

```diff
@@ -2355,6 +2390,8 @@ def compile(self, learning_rate, momentum):
             for w in self.keras_model.trainable_weights
             if 'gamma' not in w.name and 'beta' not in w.name]
         self.keras_model.add_loss(tf.add_n(reg_losses))
+        if self.config.NUm_IMAGe_CLASSES:
+            self.keras_model.add_loss("categorical_crossentropy")
```

Comment:
How does this work? Network's `add_loss` can only take tensors AFAIK. 

Also, even if there's a std function like `keras.losses.SparseCategoricalCrossentropy()(gt_image_class, image_class)` (sparse since we are passing integers), it's better to define the loss graph as a custom function, because I'm pretty sure we'd also have to mask the batch-padded samples here – like we do in the other loss graphs, e.g. https://github.com/bertsky/Mask_RCNN/blob/876e42efb0b065006ef4daa59603fc96f1c4b19e/mrcnn/model.py#L1183-L1192

The place to define and add the loss is L2191ff (i.e. your L2215ff) rather than here.

Reply:
I don't know why you think we pass the classes as integers [L1871-L1875](https://github.com/bertsky/Mask_RCNN/blob/461bbe34fa117df5979e197b19bb6ea7045ee0cc/mrcnn/model.py#L1871-L1875).